packages/dd-trace/test/opentracing/span.spec.js

```diff
@@ -163,6 +166,22 @@ describe('Span', () => {
     expect(span.context()._trace.tags['_dd.p.tid']).to.match(/^[a-f0-9]{8}0{8}$/)
   })
 
+  it('should be published via dd-trace:span:start channel', () => {
+    const onSpan = sinon.stub()
+    startCh.subscribe(onSpan)
+
+    const fields = {
+      operationName: 'operation'
+    }
+
+    span = new Span(tracer, processor, prioritySampler, fields)
+
+    expect(onSpan).to.have.been.calledOnce
+    expect(onSpan.firstCall.args[0]).to.deep.equal({ span, fields })
+
+    startCh.unsubscribe(onSpan)
```

Comment:
unsubscribe isn't called if the test fail because of an expect() throwing for example

Reply:
not sure tbh, but it's better for test isolation, it might have side effects that would be a huge pain to debug